students/csdotson/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Lesson 4 - Mailroom... part 2
+
+### Function Definitions ###
+def prompt(prompt, menu):
+    # Create menu based on arguments
+    while True:
+        response = input(prompt)
+        if menu[response]() == "quit":
+            break
+
+
+def send_thank_you():
+    # Create sub_prompt for creating thank you's and adding new donors/donations
+    prompt(thank_you_prompt, thank_you_menu)
+
+
+def add_donation():
+    # Append donation to existing donor, or add donor/donation to existing list
+    response = input("\nPlease enter a donor name: ")
+    if response not in donations.keys():
+        new_donor = response
+        new_donation = input(f"Please enter a donation amount for {new_donor}: ")
+        donations[new_donor] = [float(new_donation)]
+        print_email(new_donor, new_donation)
+    else:
+         current_donor = response
+         new_donation = input(f"Please enter a donation amount for {current_donor}: ")
+         donations[current_donor].append(float(new_donation))
+         print_email(current_donor, new_donation)
```

Comment:
I think you could eliminate some of the repeated code here (always a good thing) by doing something like this:

```
def add_donation():
    donor = input("\nPlease enter a donor name: ")
    new_donation = input(f"Please enter a donation amount for {donor}: ")
    if donor not in donations:
        donations[donor] = [float(new_donation)]
    else:
        donations[donor].append(float(new_donation))
    print_email(donor, new_donation)
```